lib/mix/lib/mix/local/installer.ex

```diff
@@ -29,33 +49,37 @@ defmodule Mix.Local.Installer do
   def install({module, name}, argv, switches) do
     {opts, args} = OptionParser.parse!(argv, strict: switches)
 
-    case args do
-      [src] ->
-        with :ok <- check_argument(src), :ok <- module.check_path_or_url(src) do
-          do_install({module, name}, src, opts)
-        else
-          {:error, message} -> Mix.raise message <> "\n" <> usage(name)
+    install_spec = parse_args(args, opts)
+
+    case module.check_install_spec(install_spec, opts) do
+      :ok -> :noop
+      {:error, message} -> Mix.raise message <> "\n" <> usage(name)
+    end
+
+    case install_spec do
+      {:fetcher, dep_spec} ->
+        fetch dep_spec, fn mixfile ->
+          case module.build(mixfile) do
+            :ok ->
+              argv = if opts[:force], do: ["--force"], else: []
+              install({module, name}, argv, switches)
+
+            {:error, reason} ->
```

Comment:
I think we don't need this branch. We can let the module.build just call `Mix.raise`. We only have this branch in check because of the usage bit, IIRC.